### PR TITLE
Warn upon too few bcrypt.kdf() rounds

### PR DESCRIPTION
**Pull request by lgommans**
Pull request as suggested for #103

### CODE REVIEW

File: src/bcrypt/__init__.py [hunk 1]
```diff
@@ -125,6 +126,14 @@ def kdf(password, salt, desired_key_bytes, rounds):
     if rounds < 1:
         raise ValueError("rounds must be 1 or more")
 
+    if rounds < 50 and not ignore_few_rounds:
+        # They probably think bcrypt.kdf()'s rounds parameter is logarithmic,
+        # expecting this value to be slow enough (it probably would be if this
+        # were bcrypt). Emit a warning.
+        sys.stderr.write(("Warning: bcrypt.kdf() called with only {} round(s). "
```
Comment (reaperhulk) [inline src/bcrypt/__init__.py]: You can do this with `warnings.warn` (https://docs.python.org/2/library/warnings.html). You can see an example here: https://github.com/pyca/cryptography/blob/a5865d9b630df7586785b683ed145b977ad7e0c4/src/cryptography/__init__.py#L22-L27

That example uses `DeprecationWarning`, but `UserWarning` is more appropriate here.

Comment (lgommans) [inline src/bcrypt/__init__.py]: I did not know of that library, thanks for the suggestion. I've added another commit to use the library. Is this better?

**Comment by lgommans** [comment]
Ah yes. I wasn't sure how to test when using sys.stderr so I left it, but forgot to check again for the warnings module.

Looking at the other kdf tests, most use only 10 rounds. Of course it makes sense in unit tests, but I hope pytest doesn't mind the warnings that it will throw.

**Comment by reaperhulk** [comment]
Tests are failing because the branch where rounds > 50 or ignore_few_rounds is true is not taken (and we have fail under 100% coverage set). Could you add a test that does that? You can verify that it all works by running tox locally via `tox -e py27` (or `py35` if you've got python 3.5 installed).

File: src/bcrypt/__init__.py [hunk 2]
```diff
@@ -112,7 +113,7 @@ def checkpw(password, hashed_password):
     return _bcrypt.lib.timingsafe_bcmp(ret, hashed_password, len(ret)) == 0
 
 
-def kdf(password, salt, desired_key_bytes, rounds):
+def kdf(password, salt, desired_key_bytes, rounds, ignore_few_rounds = False):
```
Comment (reaperhulk) [inline src/bcrypt/__init__.py]: The code linter wants no spaces around the =

File: src/bcrypt/__init__.py [hunk 3]
```diff
@@ -125,6 +126,16 @@ def kdf(password, salt, desired_key_bytes, rounds):
     if rounds < 1:
         raise ValueError("rounds must be 1 or more")
 
+    if rounds < 50 and not ignore_few_rounds:
+        # They probably think bcrypt.kdf()'s rounds parameter is logarithmic,
+        # expecting this value to be slow enough (it probably would be if this
+        # were bcrypt). Emit a warning.
+        warnings.warn((
+            "Warning: bcrypt.kdf() called with only {} round(s). "
```
Comment (reaperhulk) [inline src/bcrypt/__init__.py]: This needs to be {0} to work on python 2.6

**Comment by reaperhulk** [comment]
Jenkins, ok to test

**Comment by lgommans** [comment]
First time I went through a process like this, was interesting to see -- and it made me feel quite inept. Thanks for putting up with it, I now know better where to look to spot the issues myself!

**Comment by reaperhulk** [comment]
jenkins, retest this please

**Comment by reaperhulk** [comment]
@lgommans thanks for sticking with it, contributing to a new OSS project can be a challenge. You did a great job from my perspective :)

**Comment by lgommans** [comment]
Happy to hear that, thanks :)

**Pull request merged**